scout/utils/ensembl_rest_clients.py

```diff
@@ -76,19 +78,22 @@ def send_request(self, url):
         """
         data = {}
         try:
-            request = Request(url, headers=HEADERS)
-            response = urlopen(request)
-            content = response.read()
-            if content:
-                data = json.loads(content)
-        except HTTPError as e:
-            LOG.info("Request failed for url {0}: Error: {1}\n".format(url, e))
-            data = e
-        except ValueError as e:
-            LOG.info("Request failed for url {0}: Error: {1}\n".format(url, e))
-            data = e
+            response = requests.get(url, headers=HEADERS)
+            if response.status_code == 404:
+                LOG.info("Request failed for url %s\n", url)
+                response.raise_for_status()
+            data = response.json()
+        except requests.exceptions.MissingSchema as err:
+            LOG.info("Request failed for url %s: Error: %s\n", url, err)
+            data = err
+        except requests.exceptions.HTTPError as err:
+            LOG.info("Request failed for url %s: Error: %s\n", url, err)
+            data = err
         return data
 
+    def __repr__(self):
```

Comment:
Where is this called from?

Reply:
This is not called, it should always be included in every class. It is to create a unique representation of a class instance.

Reply:
Ah OK!